base/boot.jl

```diff
@@ -183,16 +183,23 @@ else
 end
 
 function Typeof end
-(f::typeof(Typeof))(x::ANY) = isa(x,Type) ? Type{x} : typeof(x)
+ccall(:jl_toplevel_eval_in, Any, (Any, Any),
+      Core, quote
+      (f::typeof(Typeof))(x) = ($(_expr(:meta,:nospecialize,:x)); isa(x,Type) ? Type{x} : typeof(x))
+      end)
```

Comment:
why does this still need the more complex form? Also, why does it need `f::typeof(Typeof)` – could we just use `Typeof(@nospecialize x) = ...`